src/docs/arch-row.dox

```diff
@@ -1,12 +1,68 @@
-/*! @arch_page arch-row Row Store
+/*! @arch_page arch-row-column Row Store & Column Store
 
-Row Stores are Btrees stored in WiredTiger that do not have a record id
-as key.  Thus, they implement a generalized version of a Btree, where
-the key and data can be arbitrary length.
+@section Definition
+
+A row oriented storage is a database management system where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column, the data is stored by column. WriedTiger supports both and each management system is
+represented by a certain type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. However, column-stores use a 64-bit unsigned integer record id as
+keys, thus, column-stores implement a specialized version of a B-Tree where the key has a
+predictable length.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+ is created. The fixed length column store has specialized use cases like bitmaps
+- The variable length is a more general case which allows for values to have any length, and may
+have arbitrary types, including aggregates of various types.
+
+@section Main differences
+
+Column-store keys are derived from the value's location in the table while row-store keys are
+physically stored in the table. This means there is no repetition of keys across multiple files for
+column-stores while row-store keys are duplicated usually within the same single file. Therefore,
+column-stores have a real benefit over row-stores when it comes to finding a record, since there is
+only one key per page and keys take almost no storage, fast lookups can be performed and it is easy
+to move between records. For row-stores, a slower approach is necessary (i.e multiple binary
+searches) to locate a particular record and ultimately pull together a complete row.
+
+When reading and writing column-stores, only the columns required for any particular query are
+maintained in memory. For row-stores, the memory usage will depend on the definition of column
+groups required by the query.
```

Comment:
I see where this is heading, but why would memory usage depend on the column groups required by the query? That's CS, RS loads the rows required by the query.

Reply:
I am probably misunderstanding how things work.
`RS loads the rows required by the query`
If we have columns A, B, C and D and we create multiple column groups: A/B and C/D.
If I am interested in the field in column A, can I retrieve it without loading the data associated to the columns C and D? Or we always need to "fully load" the entire row ?

Reply:
@etienneptl, I think you're correct, I think I'm wrong.